crates/matrix-sdk/src/sliding_sync/client.rs

```diff
@@ -55,6 +55,10 @@ impl SlidingSyncResponseProcessor {
 
         self.to_device_events =
             self.client.base_client().process_sliding_sync_e2ee(extensions).await?;
+
+        // Some new keys might have been received, so trigger a backup if needed.
+        self.client.encryption().backups().maybe_trigger_backup();
```

Comment:
I guess there is no need to check if it's enabled or not. It would be no op if not

Reply:
We need to check encryption is needed. We never know if it’s going to be a no-op in the future. Also we probably need to check some feature flags too.

Reply:
The enclosing method has `#[cfg(feature = "e2e-encryption")]`

Reply:
Ah true, sorry I've missed that. Well, it should be good to me then. Waiting on another review from @bnjbvr though.

Reply:
This will only work for sliding sync. How do you feel about putting this under `preprocess_to_device_events`, which is called by both sliding sync and sync v2?

Reply:
@BillCarsonFr let me know that `maybe_trigger_backup()` lives in the high-level SDK crate, while I was proposing to put it in the lower-level sdk-base crate. It's a fair point, so instead can you duplicate this line in the other code path for sync v2 please? (or if you use another mechanism like subscriptions, please re-assign me for review)

Reply:
> @BillCarsonFr let me know that `maybe_trigger_backup()` lives in the high-level SDK crate, while I was proposing to put it in the lower-level sdk-base crate. It's a fair point, so instead can you duplicate this line in the other code path for sync v2 please? (or if you use another mechanism like subscriptions, please re-assign me for review)

Done, added test for both syncs type